docs/vgclog_examples.md

```diff
@@ -0,0 +1,594 @@
+
+<exclusive-start id="2" timestamp="2020-10-18T13:27:09.603" intervalms="1913.853">
+
+<response-info timems="0.030" idlems="0.030" threads="0" lastid="0000000000AC4600" lastname="LargeThreadPool-thread-3" />
+
+</exclusive-start>
+```
+
+The `<af-start>` element indicates that the cycle was triggered by an allocation failure in the *nursery* (`type="nursery"`) area of the heap:
+
+```xml
+<af-start id="3" threadId="0000000000AC4F80" totalBytesRequested="272" timestamp="2020-10-18T13:27:09.603" intervalms="1913.921" type="nursery" />
+```
+
+The `<cycle-start>` element marks the start of the cycle. The attribute `type="scavenge"` confirms that this activity is a *scavenge* partial GC cycle:
+
+```xml
+<cycle-start id="4" type="scavenge" contextid="0" timestamp="2020-10-18T13:27:09.603" intervalms="1913.959" />
+```
+
+Most elements are labeled with an `id` attribute that increases in value incrementally, a`timestamp` attribute, and a `contextid` attribute. All elements that record GC increments and operations that are associated with a particular cycle have a `contextid` value that matches the `id` value of the cycle. The `<cycle-start>` element of this example cycle has a `id="4"`, so all subsequent elements that have a `contextid="4"`, such as the `<gc-start>` increment element and the `<gc-op>` operation element, are associated with this particular example cycle.  
+
+The `<gc-start>` element records the first GC increment. In this section, you can find information about the amount of memory available (`<mem-info>`) and where it is located in the heap.
+
+The memory snapshot within the `<gc-start>` element is taken before the *scavenge* operation and can be compared with a similar snapshot that is taken afterwards to understand the effect on the heap.
+
+```xml
+<gc-start id="5" type="scavenge" contextid="4" timestamp="2020-10-18T13:27:09.603">
+
+<mem-info id="6" free="802568640" total="1073741824" percent="74">
+
+<mem type="nursery" free="0" total="268435456" percent="0">
+
+<mem type="allocate" free="0" total="134217728" percent="0" />
+
+<mem type="survivor" free="0" total="134217728" percent="0" />
+
+</mem>
+
+<mem type="tenure" free="802568640" total="805306368" percent="99">
+
+<mem type="soa" free="762302912" total="765040640" percent="99" />
+
+<mem type="loa" free="40265728" total="40265728" percent="100" />
+
+</mem>
+
+<remembered-set count="30758" />
+
+</mem-info>
+
+</gc-start>
+
+<allocation-stats totalBytes="136862352" >
+
+<allocated-bytes non-tlh="7097008" tlh="129765344" />
+
+<largest-consumer threadName="Start Level: Equinox Container: c25c3c67-836a-49e7-9949-a16d2c2a5a4d" threadId="000000000045E300" bytes="46137952" />
+
+</allocation-stats>
+```
+
+In this example, the *nursery* area has no free space available in either the *allocate* or *survivor* spaces. The *tenure* area has 99% free in the small object area (SOA). The large object area (LOA) contains no objects.
```

Comment:
You can elaborate a bit for allocate and survivor:

At gc-start:
- allocate is close to or exactly 0% free, and that is the reason why allocation failure occurred what triggered this GC cycle
- survivor is reported exactly as 0% free (from mutator threads perpsective), although it's in fact completely free (from GC threads perspective), since it's reserved for usage during GC/scavenge time.

Later for for GC end this will hold true:
- typically a large portion (80-90%) of allocate space will be free (hence GC cycled reclaimed a large potion of allocate/nursery)
- survivor will again be 0% (while in fact is 100% free, but reserved for the next scavenge cycle)

Reply:
Is the following correct?

```xml
<gc-start id="12395" type="scavenge" contextid="12394" timestamp="2020-10-18T13:35:45.000">
  <mem-info id="12396" free="414960320" total="1073741824" percent="38">
    <mem type="nursery" free="0" total="268435456" percent="0">
      <mem type="allocate" free="0" total="241565696" percent="0" />
      <mem type="survivor" free="0" total="26869760" percent="0" />
    </mem>
    <mem type="tenure" free="414960320" total="805306368" percent="51">
      <mem type="soa" free="374694592" total="765040640" percent="48" />
      <mem type="loa" free="40265728" total="40265728" percent="100" />
    </mem>
    <remembered-set count="21474" />
  </mem-info>
</gc-start>
```

In this example:
- The *allocate* space of the *nursery* area is full, or close to full. The lack of available memory in this area triggered the allocation failure.
- The *survivor* space of the *nursery* area is reported as 'full' to reflect that there is no available memory to allocate to the mutator threads. The entire *survivor* space is instead reserved for the GC operations for the duration of the increment.
- the *nursery* area has no free space available in either the *allocate* or *survivor* spaces. The *tenure* area has 51% free.